contracts/core/src/xchain/FeeOracleV2.sol

```diff
@@ -0,0 +1,238 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity =0.8.24;
+
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { IFeeOracle } from "../interfaces/IFeeOracle.sol";
+import { IFeeOracleV2 } from "../interfaces/IFeeOracleV2.sol";
+
+/**
+ * @title FeeOracleV2
+ * @notice A simple fee oracle with a fixed fee, controlled by an admin account
+ *         Used by OmniPortal to calculate xmsg fees
+ */
+contract FeeOracleV2 is IFeeOracle, IFeeOracleV2, OwnableUpgradeable {
+    /**
+     * @notice Base gas limit for each xmsg.
+     */
+    uint64 public baseGasLimit;
```

Comment:
maybe a bit more descriptive name - I actually don't know what this is referring to. Is it the minimum gas limit a developer can set?

Reply:
No that's `xmsgMinGasLimit` in the portal .

`baseGasLimit` is the base gas limit we charge for each xmsg, to account for overhead.